store/index/index.go

```diff
@@ -949,8 +932,18 @@ func loadBucketState(ctx context.Context, basePath string, buckets Buckets, size
 		}
 	}()
 
+	fi, err := file.Stat()
+	if err != nil {
+		return err
+	}
+
+	// If the file is not the expected size then do not use it.
+	if fi.Size() != int64(types.OffBytesLen*len(buckets)) {
+		return fmt.Errorf("bucket state file is wrong size, expected %d actual %d", types.OffBytesLen*len(buckets), fi.Size())
+	}
```

Comment:
do we want to do this syscall every time we load a bucket?

Reply:
@willscott that function, `loadBucketState`,  is called only once at startup. It loads the previous bucket state saved from the previous clean shutdown, as a faster alternative to scanning the entire index to rebuild the buckets.

Reply:
The whole functional change is [here](https://github.com/ipld/go-storethehash/pull/63/files#diff-30cde1cf0cf439cb0188826c2d89b2f3a18f4e3dd5c36e8e84d56be42178ebb5R758-R761).  Everything else is just removing unused references.

Reply:
makes sense